### DISCUSSION TITLE
Curios config not loaded (conflicts ?) when jumping to server

### DISCUSSION THREAD

**Discussion by buloz** [TheIllusiveC4/Curios#384] [Q&A]
Hello,

I tried to configure curios datapack for my server. But when players logs to my server, their curios configuration seems broken and don't respect the rules of my datapack. If I reset their curios interface, it seems to fix their configuration. Is there any config file from the server I need to override to force my datapack config to be loaded by clients on logon ?

What I've done":
Since I use several mods that comes with curios configurations, I tried to override those that I'm using.
To do that I placed :
- every slots I need to tweak in data/curios/tags/items
- configure them with order & add_cosmetics options
- only replaced those I don't want to use with operation : SET, replace : true, size : 0

You can find my datapack in attachments if you want to help.

Thank you ! :)
[curios_gotham_0.2.zip](https://github.com/TheIllusiveC4/Curios/files/14364103/curios_gotham_0.2.zip)

**Comment by buloz** [comment]
Forgot to add that I've managed to lost the content of all my curios slots by logging off/logging back on the server right away and I think it's related to this config issue. I don't know if I'm facing a bug or if I just missed some guidelines of use

**Comment by TheIllusiveC4** [comment]
There's a lot of potential things going on.

First, I want to make sure of one thing: Are you using this in a pure Forge server or are you using this with any plugins on a hybrid server like Magma?

> But when players logs to my server, their curios configuration seems broken and don't respect the rules of my datapack. If I reset their curios interface, it seems to fix their configuration.

In what way exactly are they broken? Are slots missing or just incorrect or both? How do you reset their interface and what is the correct behavior?

> every slots I need to tweak in data/curios/tags/items

This is incorrect. The `data/curios/tags/items` folder are for tag files, which should only have `"values"` of items for the tag and optionally `"replace"`. This is not where you place slot settings for Curios. Slot settings should be entirely within your `data/curios_gotham/curios/slots` folder.

**Comment by buloz** [comment]
I use an Arclight server, so hybrid.
Attached screenshot will show in which way slots are broken : they aren't at the same order, also if I try to place an item in its slot (i.e. backpack I added in back.json), it will appear in the curios inventory but won't be usable. If I put it in the slot at the correct position (i.e. : back is supposed to be in 2nd position, when I log in then hands are in 2nd position), I am able to equip backpack in my hands and somehow it duplicates the item with making it appears in the back slot too.
I think I managed to make an item void by doing this too.
I use /curios reset to have the correct behavior, but everytime a player logs off then when they logs in they remains with the broken state.
I'll try with the correct way to place attributes in json files : 
values in data/curios/tags/items then behavior in data/{namespace}/curios/slots

Thank you for your answer.

Incorrect :
![before_reset](https://github.com/TheIllusiveC4/Curios/assets/45174948/f862db5b-65e4-48a7-953f-57e83b23ffd0)

Correct :
![after_reset](https://github.com/TheIllusiveC4/Curios/assets/45174948/c9bab91f-6ffa-49e1-910e-26e65ece68c3)

**Comment by buloz** [comment]
I can confirm I have the same problem with this datapack :
[curios_gotham_0.3.zip](https://github.com/TheIllusiveC4/Curios/files/14366538/curios_gotham_0.3.zip)

**Comment by TheIllusiveC4** [comment]
> I use an Arclight server, so hybrid.

Historically, when there is odd slot behavior, it is usually due to this hybrid environment or your installed plugins. Attempting to load mods and plugins together may result in random bugs such as this and it is hard to diagnose because mods and plugins were not designed in the same environment. Unfortunately, I cannot provide much further insight or support into this unless you have this problem on a purely Forge server.

**Reply by buloz** [reply]
Ok thank you, I'll occasionally try it one day on a forge only server and stay you tune
